operator/controllers/resources/github_connector_controller_test.go

```diff
@@ -0,0 +1,128 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/gravitational/trace"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gravitational/teleport/api/types"
+	resourcesv3 "github.com/gravitational/teleport/operator/apis/resources/v3"
+)
+
+var githubSpec = types.GithubConnectorSpecV3{
+	ClientID:      "client id",
+	ClientSecret:  "client secret",
+	RedirectURL:   "https://redirect",
+	TeamsToLogins: nil,
+	Display:       "",
+	TeamsToRoles: []types.TeamRolesMapping{{
+		Organization: "test",
+		Team:         "test",
+		Roles:        []string{"test"},
+	}},
+}
+
+type githubTestingPrimitives struct {
+	setup *testSetup
+}
+
+func (g *githubTestingPrimitives) init(setup *testSetup) {
+	g.setup = setup
+}
+
+func (g *githubTestingPrimitives) setupTeleportFixtures(ctx context.Context) error {
```

Comment:
The linter sees this as unused, I think this is due to generics. Will test with golang-ci's latest version to see if this helps

Reply:
Latest golangci-lint release does not solve the issue. I'll disable `unused` on those lines in the meantime: https://github.com/dominikh/go-tools/issues/1294

Reply:
Added a linter exclusion rule